c_api/blockingconcurrentqueue.cpp

```diff
@@ -0,0 +1,48 @@
+#include "concurrentqueue.h"
+#include "../blockingconcurrentqueue.h"
+
+typedef moodycamel::BlockingConcurrentQueue<void*> MoodycamelBCQType, *MoodycamelBCQPtr;
+
+extern "C" {
+	
+int moodycamel_bcq_create(MoodycamelBCQHandle* handle)
+{
+	if (handle == nullptr) {
+		return 0;
+	}
+	MoodycamelBCQPtr retval = new MoodycamelBCQType;
+	if (retval == nullptr) {
+		return 0;
+	}
+	*handle = retval;
+	return 1;
+}
+
+int moodycamel_bcq_destroy(MoodycamelBCQHandle handle)
+{
+	delete reinterpret_cast<MoodycamelBCQPtr>(handle);
+	return 1;
+}
+
+int moodycamel_bcq_enqueue(MoodycamelBCQHandle handle, MoodycamelValue value)
+{
+	if (handle == nullptr) {
+		return 0;
+	}
+	reinterpret_cast<MoodycamelBCQPtr>(handle)->enqueue(value);
+	return 1;
+}
+
+int moodycamel_bcq_wait_dequeue(MoodycamelBCQHandle handle, MoodycamelValue* value)
```

Comment:
we also need a plain bcq_try_dequeue or there's no way to drain the queue without knowing in advance how many elements are in it